domain/access/service/permission_test.go

```diff
@@ -93,15 +93,17 @@ func (s *serviceSuite) TestUpsertPermission(c *gc.C) {
 	err := NewPermissionService(s.state).UpsertPermission(
```

Comment:
I tend to avoid letting database jargon escape from the database layer. So `UpsertPermission` is fine for state, but in a service, I would call it `GrantPermission` or something similar. Same treatment for the args struct.

Reply:
It covers Grant and Revoke, so `UpdatePermission`.